File: bogosort.rb
def shuffle(arr)
    i = arr.length - 1
    while i > 0
        r = rand(i)
        arr[i], arr[r] = arr[r], arr[i]
        i -= 1
    end
    arr
end

def bogosort(arr)
    arr = shuffle(arr) until arr == arr.sort
    arr
end

File: bogosort.c
#include <time.h>
#include <stdlib.h>

void shuffle(int*& arr)
{
    int len = sizeof(arr) / sizeof(arr[0]);
	
    if (len > 1) 
    {
        for (int i = len - 1; i > 0; i--) 
        {
            int temp = arr[i];
            int r = rand() % (len + 1);
            arr[i] = arr[r];
            arr[r] = temp;
        }        
    }
}

void bogosort(int*& arr, int* order)
{
  while (arr != order) 
	{
    shuffle(arr);
	}
}
